Classes/Middleware/DOMDocumentValidation.php

```diff
@@ -71,30 +76,31 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         // check required parameters
         $urlParam = $parameters['url'];
         if (!isset($urlParam)) {
-            throw new InvalidArgumentException('URL parameter is missing.', 1724334674);
+            return $this->getJsonResponse('URL parameter is missing.', self::BAD_REQUEST);
         }
 
         $typeParam = $parameters['type'];
         if (!isset($typeParam)) {
-            throw new InvalidArgumentException('Type parameter is missing.', 1744373423);
+            return $this->getJsonResponse('Type parameter is missing.', self::BAD_REQUEST);
         }
-
         // load dom document from url
         if (!GeneralUtility::isValidUrl($urlParam)) {
-            $this->logger->debug('Parameter "' . $urlParam . '" is not a valid url.');
-            throw new InvalidArgumentException('Value of url parameter is not a valid url.', 1724852611);
+            return $this->getJsonResponse('Parameter "' . $urlParam . '" is not a valid url.', self::NOT_FOUND);
```

Comment:
```suggestion
            return $this->getJsonResponse('Parameter "' . $urlParam . '" is not a valid URL.', self::NOT_FOUND);
```
We should use uppercase letters for "URL" consistently. ;)